internal/actions/reward.go

```diff
@@ -7,5 +7,5 @@ import (
 
 // RewardNode updates LastUsed for provided node
 func (a actions) RewardNode(node models.Node, repositories repositories.Repos) {
```

Comment:
Naming of this kinda does not make sense now

Reply:
Shouldn't RewardNode save node record or something

Reply:
> Shouldn't RewardNode save node record or something

We have separate functions that handle failed and successful request (and are saving node records), but I agree that this naming doesn't have sense anymore.

```go
// FailedRequest should be called when rpc response is invalid to penalize node.
// It does not return value as it should be called in separate goroutine
func FailedRequest(node models.Node, repositories repositories.Repos, actions actions.Actions) {
    actions.PenalizeNode(node, repositories)

    err := repositories.RecordRepo.Save(&models.Record{
        NodeId:    node.ID,
        Timestamp: time.Now(),
        Status:    "failed",
    })
    if err != nil {
        log.Errorf("Failed saving failed request because of: %v", err)
    }

    log.Debugf("Node %s failed to serve successful request", node.ID)
}

// SuccessfulRequest should be called when rpc response is valid to reward node.
// It does not return value as it should be called in separate goroutine
func SuccessfulRequest(node models.Node,  repositories repositories.Repos, actions actions.Actions) {
    actions.RewardNode(node, repositories)

    err := repositories.RecordRepo.Save(&models.Record{
        NodeId:    node.ID,
        Timestamp: time.Now(),
        Status:    "successful",
    })
    if err != nil {
        log.Errorf("Failed saving successful request because of: %v", err)
    }

    log.Debugf("Node %s served successful request", node.ID)
}
```

@mpetrun5 I guess I can even remove this **RewardNode** action and call directly node repo